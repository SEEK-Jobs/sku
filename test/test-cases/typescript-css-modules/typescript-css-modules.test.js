const path = require('path');
const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');

describe('typescript-css-modules', () => {
  describe('build', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(distDir);
      expect(files).toMatchSnapshot();
    });
  });

  describe('build-ssr', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build-ssr', appDir);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(distDir);
      expect(files).toMatchSnapshot();
    });
  });

  describe('test', () => {
    let exitCode;

    beforeAll(async () => {
      const { childProcess } = await runSkuScriptInDir('test', appDir);
      exitCode = childProcess.exitCode;
    });

    it('should handle Less and css.js in tests', async () => {
      expect(exitCode).toEqual(0);
    });
  });

  describe('lint', () => {
    let exitCode;

    beforeAll(async () => {
      // run build first to ensure typescript declarations are generated
      await runSkuScriptInDir('build', appDir);
      const { childProcess } = await runSkuScriptInDir('lint', appDir);
      exitCode = childProcess.exitCode;
    });

    it('should handle tsc and tslint', async () => {
      expect(exitCode).toEqual(0);
    });
  });
});
