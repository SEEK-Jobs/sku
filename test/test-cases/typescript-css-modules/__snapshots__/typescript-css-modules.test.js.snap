// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript-css-modules build should create valid app 1`] = `
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="http://localhost:4003/main-bfe8911084b821bf8edd.css"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/runtime-62bc0856cbb8d47f9471.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/vendors~main-a264bd1d24dfb67ae5a1.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/main-0b09f9bf338f7b37cc2d.js"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div class="_1brbpTo Jlnxw97">
        <div class="_3YtmnHj lwsthgY"
             data-automation-text="true"
        >
          Hello World
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/runtime-62bc0856cbb8d47f9471.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/vendors~main-a264bd1d24dfb67ae5a1.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/main-0b09f9bf338f7b37cc2d.js"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`typescript-css-modules build should generate the expected files 1`] = `
Object {
  "components/BlueBlock/BlueBlock.less.d.ts": "// This file is automatically generated.
// Please do not change this file!
export const border: string;
export const root: string;
",
  "components/RedBlock/RedBlock.less.d.ts": "// This file is automatically generated.
// Please do not change this file!
export const border: string;
export const root: string;
",
  "index.html": <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="http://localhost:4003/main-bfe8911084b821bf8edd.css"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/runtime-62bc0856cbb8d47f9471.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/vendors~main-a264bd1d24dfb67ae5a1.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/main-0b09f9bf338f7b37cc2d.js"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div class="_1brbpTo Jlnxw97">
        <div class="_3YtmnHj lwsthgY"
             data-automation-text="true"
        >
          Hello World
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/runtime-62bc0856cbb8d47f9471.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/vendors~main-a264bd1d24dfb67ae5a1.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/main-0b09f9bf338f7b37cc2d.js"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>,
  "jsStyles.css.js.d.ts": "// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'nested': string;
  'root': string;
}
declare const cssExports: CssExports;
export = cssExports;
",
  "lessStyles.less.d.ts": "// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'nested': string;
  'root': string;
}
declare const cssExports: CssExports;
export = cssExports;
",
  "main-0b09f9bf338f7b37cc2d.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main-bfe8911084b821bf8edd.css": ._1brbpTo {
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}
._1brbpTo ._3YtmnHj {
  font-size: 32px;
}
.Jlnxw97 {
  -webkit-align-items: center;
  -ms-flex-align: center;
  align-items: center;
}
.Jlnxw97 .lwsthgY {
  color: red;
}
,
  "runtime-62bc0856cbb8d47f9471.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-a264bd1d24dfb67ae5a1.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`typescript-css-modules build-ssr should create valid app 1`] = `
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="http://localhost:4003/main-bfe8911084b821bf8edd.css"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/runtime-62bc0856cbb8d47f9471.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/vendors~main-984fab7243c3dc553e6f.js"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="http://localhost:4003/main-fe572733ec996a636549.js"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div class="_1brbpTo Jlnxw97">
        <div class="_3YtmnHj lwsthgY"
             data-automation-text="true"
        >
          Hello World
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/runtime-62bc0856cbb8d47f9471.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/vendors~main-984fab7243c3dc553e6f.js"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="http://localhost:4003/main-fe572733ec996a636549.js"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`typescript-css-modules build-ssr should generate the expected files 1`] = `
Object {
  "components/BlueBlock/BlueBlock.less.d.ts": "// This file is automatically generated.
// Please do not change this file!
export const border: string;
export const root: string;
",
  "components/RedBlock/RedBlock.less.d.ts": "// This file is automatically generated.
// Please do not change this file!
export const border: string;
export const root: string;
",
  "jsStyles.css.js.d.ts": "// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'nested': string;
  'root': string;
}
declare const cssExports: CssExports;
export = cssExports;
",
  "lessStyles.less.d.ts": "// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'nested': string;
  'root': string;
}
declare const cssExports: CssExports;
export = cssExports;
",
  "main-bfe8911084b821bf8edd.css": ._1brbpTo {
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}
._1brbpTo ._3YtmnHj {
  font-size: 32px;
}
.Jlnxw97 {
  -webkit-align-items: center;
  -ms-flex-align: center;
  align-items: center;
}
.Jlnxw97 .lwsthgY {
  color: red;
}
,
  "main-fe572733ec996a636549.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-62bc0856cbb8d47f9471.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "server.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-984fab7243c3dc553e6f.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`typescript-css-modules start should start a development server 1`] = `
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="/main.css"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/runtime.js"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/vendors~devServerOnly~main.js"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/vendors~main.js"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="/main.js"
    >
  </head>
  <body>
    <div id="app">
      <div class="lessStyles__root___1brbpTo jsStyles-css__root___Jlnxw97">
        <div class="lessStyles__nested___3YtmnHj jsStyles-css__nested___lwsthgY"
             data-automation-text="true"
        >
          Hello World
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="/runtime.js"
    >
    </script>
    <script async
            data-chunk="main"
            src="/vendors~devServerOnly~main.js"
    >
    </script>
    <script async
            data-chunk="main"
            src="/vendors~main.js"
    >
    </script>
    <script async
            data-chunk="main"
            src="/main.js"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
